src/map/status.cpp

```diff
@@ -13960,6 +13966,18 @@ TIMER_FUNC(status_change_timer){
 			return 0;
 		}
 		break;
+		
+	case SC_GRADUAL_GRAVITY:
+		if (--(sce->val4) >= 0) {
```

Comment:
```suggestion
		if (sce->val4 >= 0) {
```
For interval based SCs you only need to check if there was still time remaining.